src/arraymath.jl

```diff
@@ -149,4 +149,25 @@ end
         return a
     end
 end
+
+"""
+    arithmetic_closure(T)
+
+Return a "bigger" type, which is closed under `+,-,*,/`.
```

Comment:
"Return a type which can represent `T` and is closed under arithmetic operations `+`, `-`, `*` and `/`."

Reply:
(This is is somewhat a simplification, but let's keep it grounded.)

Reply:
Return the type which values of type `T` will promote to under a combination of
the arithmetic operations `+`, `-`, `*` and `/`.

?